app/Core/Support/DateTimeHelper.php

```diff
@@ -293,19 +293,16 @@ public function convertDateTime(
         $dateTime = trim($dateTime);
 
         if ($this->isValidDateString($dateTime)) {
-            $fromDateTimeZone = new \DateTimeZone($fromTz);
+            $fromDateTimeZone = new DateTimeZone($fromTz);
 
             //If it is an API Request, don't do any formatting, return ATOM in target tz
             if ($this->apiRequest->isApiRequest()) {
                 $utcDate = new DateTime($dateTime, $fromDateTimeZone);
                 return is_object($utcDate) ? $utcDate->format(DateTime::ATOM) : false;
             }
+            $timestamp = DateTime::createFromFormat($fromFormat, $dateTime, new DateTimeZone($toTz));
```

Comment:
It seems that the "fromTimezone" entry is not being taken into consideration anymore (unless it's an API Request). Which means this would fail for a any date save objects from other forms. 

The goal of this method is to convert a date from one timezone+format into a different timezone+format. It's deliberately made to be very flexible. We don't want to rely on the "default" timezone set by the server since different users can set different timezones. 

I think the issue with the timesheets is probably somewhere in the timesheet class rather than in the datetime helper.